File: 2024/day-03.pl
use v5.38;

open my $fh, '<', './input/day-03-input-test.txt', or die $!;

## Part 1
my $result = 0;
while (<$fh>) {
    $result += $1 * $2 while /mul\((\d+),(\d+)\)/g;
}
say $result;

File: 2015/c/day-03.c
#include <stdio.h>
#include <stdlib.h>
#include "uthash.h"
#include "utarray.h"
#include "helpers.h"

char *input = "../input/day-03.txt";

typedef struct {
    int x, y;
} House;


int main()
{
    FILE *file = fopen(input, "r");
    char line[256];

    if (file == NULL) {
        printf("Unable to open file.");
        return 1;
    }

    while (fgets(line, sizeof(line), file)) {
    }

    fclose(file);

    return 0;
}
